task/06-conditions-n-loops-tasks.js

```diff
@@ -293,13 +320,24 @@ function getDigitalRoot(num) {
  *   '[[][][[]]]' => true
  *   '[[][]][' => false
  *   '{)' = false
- *   '{[(<{[]}>)]}' = true 
+ *   '{[(<{[]}>)]}' = true
  */
 function isBracketsBalanced(str) {
-    throw new Error('Not implemented');
+  const result = [];
+  const openBrackets = ["(", "[", "{", "<"];
+  const closeBrackets = [")", "]", "}", ">"];
+  for (let i of str) {
+    if (openBrackets.indexOf(i) > -1) result.push(i);
+    else if (
+      closeBrackets.indexOf(i) ===
+      openBrackets.indexOf(result[result.length - 1])
+    )
+      result.pop();
+    else result.push(i);
```

Comment:
It's more readable when you left curly braces in all cases. Even statement is only one line